src/test/java/org/mule/test/http/functional/requester/AbstractHttpRetryRequestTestCase.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.test.http.functional.requester;
+
+import static java.lang.reflect.Modifier.FINAL;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertThat;
+import static org.mule.extension.http.internal.request.HttpRequester.DEFAULT_RETRY_ATTEMPTS;
+import static org.mule.extension.http.internal.request.HttpRequester.REMOTELY_CLOSED;
+import org.mule.extension.http.internal.request.HttpRequester;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.test.http.functional.AbstractHttpTestCase;
+import org.mule.test.http.utils.TestServerSocket;
+
+import java.lang.reflect.Field;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.internal.matchers.ThrowableMessageMatcher;
+import org.junit.rules.ExpectedException;
+
+public abstract class AbstractHttpRetryRequestTestCase extends AbstractHttpTestCase {
+
+  private static ThrowableMessageMatcher REMOTELY_CLOSE_CAUSE_MATCHER =
+      new ThrowableMessageMatcher<>(containsString(REMOTELY_CLOSED));
+
+  @Rule
+  public DynamicPort port = new DynamicPort("httpPort");
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Override
+  protected String getConfigFile() {
+    return "http-retry-policy-config.xml";
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    expectedException.expectCause(REMOTELY_CLOSE_CAUSE_MATCHER);
```

Comment:
I don't understand why you are setting the expected exception in a before. It should be used right before the method call that will fail.

Reply:
All the tests share the same exception cause. I thought it would be a good manner to avoid the same line in all the tests.

Reply:
The thing is that if something else you don't expect to fail does it will be covered by that matcher instead of simply bubbling up.